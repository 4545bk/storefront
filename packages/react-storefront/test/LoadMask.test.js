/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import LoadMask from '../src/LoadMask'

describe('LoadMask', () => {
  it('should render without children', () => {
    expect(mount(<LoadMask/>)).toMatchSnapshot()
  })

  it('should render with children', () => {
    expect(mount(<LoadMask>Loading...</LoadMask>)).toMatchSnapshot()
  })

  it('should set overflow to hidden when fullscreen==true', () => {
    const wrapper = mount(<LoadMask show fullscreen/>)
    expect(document.body.style.overflow).toBe('hidden')
    wrapper.unmount()
    expect(document.body.style.overflow).toBe('visible')
  })
})