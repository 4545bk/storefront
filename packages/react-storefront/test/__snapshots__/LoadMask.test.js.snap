// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadMask should render with children 1`] = `
<WithStyles(LoadMask)>
  <LoadMask classes={{...}} show={false} fullscreen={false}>
    <div style={[undefined]} className="RSFLoadMask-root-1">
      Loading...
    </div>
  </LoadMask>
</WithStyles(LoadMask)>
`;

exports[`LoadMask should render without children 1`] = `
<WithStyles(LoadMask)>
  <LoadMask classes={{...}} show={false} fullscreen={false}>
    <div style={[undefined]} className="RSFLoadMask-root-1">
      <WithStyles(CircularProgress) className={[undefined]} color="secondary">
        <CircularProgress classes={{...}} className={[undefined]} color="secondary" size={40} thickness={3.6} value={0} variant="indeterminate">
          <div className="MuiCircularProgress-root-4 MuiCircularProgress-colorSecondary-8 MuiCircularProgress-indeterminate-6" style={{...}} role="progressbar">
            <svg className="MuiCircularProgress-svg-9" viewBox="0 0 50 50">
              <circle className="MuiCircularProgress-circle-10 MuiCircularProgress-circleIndeterminate-12" style={{...}} cx={25} cy={25} r={20} fill="none" strokeWidth={3.6} />
            </svg>
          </div>
        </CircularProgress>
      </WithStyles(CircularProgress)>
    </div>
  </LoadMask>
</WithStyles(LoadMask)>
`;
